pkg/compact/compact_test.go

```diff
@@ -205,6 +205,148 @@ func createBlockMeta(id uint64, minTime, maxTime int64, labels map[string]string
 	return m
 }
 
+func TestRetentionProgressCalculate(t *testing.T) {
+	logger := log.NewNopLogger()
+	reg := prometheus.NewRegistry()
+	unRegisterer := &receive.UnRegisterer{Registerer: reg}
```

Comment:
I believe we don't need to use unregisterer. Can you also remove it in the compact and downsample tests?

Reply:
Thanks!
Could you PTAL at my latest commit and let me know if that's what's expected?